lib/src/main/java/com/shopify/checkoutsheetkit/CheckoutDialog.kt

```diff
@@ -148,9 +147,38 @@ internal class CheckoutDialog(
         findViewById<ProgressBar>(R.id.progressBar).visibility = visibility
     }
 
-    internal fun closeCheckoutDialogWithError(error: CheckoutException) {
-        checkoutEventProcessor.onCheckoutFailed(error)
-        dismiss()
+    internal fun closeCheckoutDialogWithError(exception: CheckoutException) {
+        if (ShopifyCheckoutSheetKit.configuration.errorRecovery.shouldRecoverFromError(exception)) {
+            attemptToRecoverFromError(exception)
+        } else {
+            checkoutEventProcessor.onCheckoutFailed(exception)
+            dismiss()
```

Comment:
Has this been the default for a while?

Reply:
You mean dismissing the dialog and calling onCheckoutFailed()?

I think so yeah, is there a difference compared with swift?